### DISCUSSION TITLE
Undefined method 'hasRole'

### DISCUSSION THREAD

**Discussion by KNMoise** [spatie/laravel-permission#2614] [Q&A]
In my source codes i ahve tried all the possible documentation to check where the error is comming from but still facing the issue after all? here are my source codes 

```php
<?php

namespace App\Http\Controllers\Auth;

use Illuminate\Support\Facades\Auth;
use Illuminate\Support\Facades\Hash; 
use App\Http\Controllers\Controller;
use Illuminate\Foundation\Auth\AuthenticatesUsers;
use Spatie\Permission\Traits\HasRoles;

class LoginController extends Controller
{
    use AuthenticatesUsers, HasRoles; // Fix: Add HasRoles trait

    protected $redirectTo = '/home';

    protected function redirectTo()
    {
        if (Auth::user()->hasRole('admin')) { // Fix: Use the correct case for Auth
            return '/admin/dashboard';
        } elseif (Auth::user()->hasRole('finance')) {
            return '/finance/dashboard';
        } elseif (Auth::user()->hasRole('stock')) {
            return '/stock/dashboard';
        } elseif (Auth::user()->hasRole('production')) {
            return '/production/dashboard';
        } else {
            return '/employee/dashboard';
        }
    }


}
`

please help to fix the issue ?

**Comment by parallels999** [comment]
You forgot adding `hasRoles` trait

**Comment by drbyte** [comment]
You probably need to inspect what roles your User has at that point.

```php
$user = Auth::user();
dd($user, $user->roles);
```

**Comment by KNMoise** [comment]
This seems to work with me, after all research and work I tried with this function then boom it worked out (auth()->check() && auth()->$user->hasRole('production'))
you may check guys on your side if it work perfect too?

**Reply by KNMoise** [reply]
if (auth()->check() && auth()->$user->hasRole('production')) {
}

**Reply by erikn69** [reply]
https://github.com/spatie/laravel-permission/blob/0e593bb9e6808cc55c67f150878b0847f20b9036/src/PermissionServiceProvider.php#L122

**Reply by KNMoise** [reply]
I have made it and fix the error am not sure to test this one again i will let you know once i have made some testes

**Reply by KNMoise** [reply]
The one i have made has brought back the same error
